lib/galaxy/tools/extract_dataset.xml

```diff
@@ -36,7 +36,7 @@
     </conditional>
   </inputs>
   <outputs>
-    <data name="output" label="${on_string} (extracted element)" />
+    <data name="output" label="${on_string} (extracted element)" format_source="input" />
```

Comment:
Do this work correctly if the input collection is not homogeneous?

Reply:
I'm not sure at all. In https://github.com/galaxyproject/galaxy/pull/11754 we try to add something in this direction https://github.com/galaxyproject/galaxy/blob/26d94faf8296dcaecdbb0034d6ec2dbf69563dce/lib/galaxy/job_execution/output_collect.py#L120 .. but I'm also unsure if this change would apply for data set outputs. 

Seems that we do not have a test for this (`collection` and `multiple="true"`). 

This looks fancy: https://github.com/galaxyproject/galaxy/blob/e149d9d32e1bca6c07c38b1a9cdabfee60323610/test/functional/tools/output_format_collection.xml#L11 .. but I guess we can not use tool parameters in `format_source`, or?

Reply:
I created a small test tool: `format_source` just takes the format of the 1st element (as far as I can see).

Could create a PR with this tool?